cg/apps/tb/api.py

```diff
@@ -66,17 +66,16 @@ def is_analysis_completed(self, case_id: str) -> bool:
 
     def has_analysis_started(self, case_id: str) -> bool:
         """Check if analysis has started"""
-        if self.is_analysis_ongoing(case_id=case_id):
-            LOG.warning("%s: analysis is ongoing - skipping", case_id)
-            return True
-
-        if self.is_analysis_failed(case_id=case_id):
-            LOG.warning("%s: analysis failed previously - skipping", case_id)
-            return True
-
-        if self.is_analysis_completed(case_id=case_id):
-            LOG.warning("%s: analysis completed - skipping", case_id)
-            return True
+        statuses = ("ongoing", "failed", "completed")
+        get_analysis_status = {
+            "ongoing": self.is_analysis_ongoing,
+            "failed": self.is_analysis_failed,
+            "completed": self.is_analysis_completed,
+        }
+        for status in statuses:
+            has_started = get_analysis_status[status](case_id=case_id)
+            if has_started:
+                return has_started, status
```

Comment:
```suggestion
           return self.is_analysis_ongoing(case_id=case_id) or self.is_analysis_failed(case_id=case_id) or self.is_analysis_completed(case_id=case_id)
```

Reply:
We decided to keep the dispatch, but let it have single responisbility of returning bool